tutorials/math/principal.py

```diff
@@ -0,0 +1,88 @@
+## \file
+##    \ingroup tutorial_math
+## \notebook
+## Principal Components Analysis (PCA) example
+##
+## Example of using TPrincipal as a stand alone class.
+##
+## We create n-dimensional data points, where c = trunc(n / 5) + 1
+## are  correlated with the rest n - c randomly distributed variables.
+##
+## \macro_output
+## \macro_code
+##
+## \authors Juan Fernando Jaramillo Botero
+##
+## based on principal.C by Rene Brun and Christian Holm Christensen
+
+from ROOT import TPrincipal, gRandom, TBrowser
+import numpy
+
+
+def principal(n=10, m=10000):
```

Comment:
Python tutorials do not aim to be importable and runable. I would avoid to enclose everything in a function.

Reply:
Done, I remove the def, and the if __name__ == '__main__' lines.